dali/pipeline/util/bounding_box.h

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_BOUNDING_BOX_H
+#define DALI_BOUNDING_BOX_H
+
+#include "dali/error_handling.h"
+
+namespace dali {
+
+class BoundingBox {
+ public:
+  const static size_t kSize = 4;
+
+  BoundingBox(const BoundingBox& other)
+      : left_{other.left_},
+        top_{other.top_},
+        right_{other.right_},
+        bottom_{other.bottom_},
+        area_{other.area_} {}
+
+  BoundingBox(BoundingBox&& other) noexcept
+      : left_{other.left_},
+        top_{other.top_},
+        right_{other.right_},
+        bottom_{other.bottom_},
+        area_{other.area_} {}
+
+  BoundingBox& operator=(BoundingBox other) {
+    left_ = other.left_;
+    top_ = other.top_;
+    right_ = other.right_;
+    bottom_ = other.bottom_;
+    area_ = other.area_;
+
+    return *this;
+  }
+
+  static BoundingBox FromLtrb(float l, float t, float r, float b) {
```

Comment:
I added a set of functions that take an array of size 4, since the size is fixed, whereas with a begin and end iterator, the main benefit is to support variable length in the container.

Nevertheless, imho the best solution would be to use a range (begin, kSize) the same way that gsl::span would work. I finally chose this way because it would make less verbose the majority of the usages of BoundingBox, where we read from data() pointers in tensors.

Reply:
https://stackoverflow.com/questions/1328223/when-a-function-has-a-specific-size-array-parameter-why-is-it-replaced-with-a-p

Reply:
Now it is https://lkml.org/lkml/2015/9/3/428

Reply:
Now it is https://lkml.org/lkml/2015/9/3/428

Reply:
accepted